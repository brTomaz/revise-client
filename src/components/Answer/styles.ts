import styled, { css } from 'styled-components'
import media from 'styled-media-query'

export const Wrapper = styled.li`
  ${({ theme }) => css`
    cursor: pointer;
    list-style: none;
    background: ${theme.colors.mainBg};
    transition: all 0.5s ease-in-out;
    color: ${theme.colors.white};
    font-weight: ${theme.font.bold};
    border-radius: 4px;

    padding: ${theme.spacings.xsmall} ${theme.spacings.xsmall};

    &:hover {
      color: ${theme.colors.secondary};
    }

    font-size: ${theme.font.sizes.large};

    ${media.greaterThan('medium')`
      font-size: ${theme.font.sizes.xlarge};
    `}
  `}
`
