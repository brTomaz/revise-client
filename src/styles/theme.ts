export default {
  border: {
    radius: '2.6rem'
  },
  font: {
    family:
      "Poppins, -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif",
    light: 300,
    normal: 400,
    bold: 600,
    sizes: {
      xsmall: '1.2rem',
      small: '1.4rem',
      medium: '1.6rem',
      large: '1.8rem',
      xlarge: '2.0rem',
      xxlarge: '2.8rem',
      huge: '3.2rem',
      xhuge: '4.0rem',
      xxhuge: '4.8rem'
    }
  },
  colors: {
    primary: '#2D3D88',
    secondary: '#2DC2C3',
    darkSecondary: '#166061',
    lightBlue: '#6493EE',
    mainBg: '#06092B',
    lightBg: '#F2F2F2',
    white: '#FAFAFA',
    black: '#030517',
    lightGray: '#EAEAEA',
    gray: '#DADADA',
    darkGray: '#3A3A3A',
    error: '#F74446'
  },
  spacings: {
    xxsmall: '0.8rem',
    xsmall: '1.6rem',
    small: '2.4rem',
    medium: '3.2rem',
    large: '4.0rem',
    xlarge: '4.8rem',
    xxlarge: '5.6rem',
    huge: '6.4rem',
    xhuge: '7.2rem',
    xxhuge: '9.6rem'
  },
  layers: {
    base: 10,
    menu: 20,
    overlay: 30,
    modal: 40,
    alwaysOnTop: 50
  }
} as const
