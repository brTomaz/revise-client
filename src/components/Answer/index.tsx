import { MouseEvent } from 'react'
import * as Styles from './styles'

export type AnswerProps = {
  answer: string
  handleAnswerClick: (event: MouseEvent) => void
}

const Answer = ({ answer, handleAnswerClick }: AnswerProps) => (
  <Styles.Wrapper onClick={handleAnswerClick}>{answer}</Styles.Wrapper>
)

export default Answer
