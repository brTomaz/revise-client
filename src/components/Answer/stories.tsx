/* eslint-disable @typescript-eslint/no-empty-function */
import { Story, Meta } from '@storybook/react/types-6-0'

import Answer, { AnswerProps } from '.'

export default {
  title: 'Answer',
  component: Answer,
  parameters: {
    backgrounds: {
      default: 'light'
    }
  }
} as Meta

export const Default: Story<AnswerProps> = (args) => <Answer {...args} />

Default.args = {
  answer:
    'O entendimento de que a luz precisa de um meio de propagação, difundido pelos defensores da existência do éter.',
  handleAnswerClick: () => {}
}
