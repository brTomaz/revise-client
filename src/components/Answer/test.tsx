import { screen } from '@testing-library/react'

import theme from 'styles/theme'
import { renderWithTheme } from 'utils/tests/helpers'
import Answer from '.'

describe('<Answer />', () => {
  let listItem: HTMLElement
  beforeEach(() => {
    renderWithTheme(
      // eslint-disable-next-line @typescript-eslint/no-empty-function
      <Answer handleAnswerClick={() => {}} answer={'any_answer'} />
    )

    listItem = screen.getByRole('listitem')
  })

  it('should render the Answer with styles', () => {
    expect(listItem).toHaveStyle({
      'font-size': theme.font.sizes.large,
      color: theme.colors.white,
      padding: `${theme.spacings.xsmall} ${theme.spacings.xsmall}`,
      'border-radius': '4px',
      'list-style': 'none'
    })

    expect(listItem).toMatchSnapshot()
  })

  it('should change color when hovering', () => {
    expect(listItem).toHaveStyleRule('color', theme.colors.secondary, {
      modifier: ':hover'
    })
  })
})
