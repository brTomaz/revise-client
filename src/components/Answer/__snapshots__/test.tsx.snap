// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Answer /> should render the Answer with styles 1`] = `
.c0 {
  cursor: pointer;
  list-style: none;
  background: #040923;
  -webkit-transition: all 0.5s ease-in-out;
  transition: all 0.5s ease-in-out;
  color: #FAFAFA;
  font-weight: 600;
  border-radius: 4px;
  padding: 1.6rem 1.6rem;
  font-size: 1.8rem;
}

.c0:hover {
  color: #2DC2C3;
}

@media (min-width:768px) {
  .c0 {
    font-size: 2.0rem;
  }
}

<li
  class="c0"
>
  any_answer
</li>
`
